File: java/leetcode/java_lib/src/main/java/java_lib/IsPalindrome/IsPalindrome.java
package java_lib.IsPalindrome;

class IsPalindrome {
	public boolean isEven;
	int numDigits;

	public boolean run(int x) {
		if (x < 0) {return false;}
		return false;
	}

	public void digits_to_reverse(int pal) {
		numDigits = (int)Math.log10(pal);
		isEven = !(0 == numDigits % 2);
		if (isEven) {numDigits += 1;}
		numDigits /= 2;
	}

	public int partial_reverse(int forward) {
		int reversed = 0;

		while (forward != 0) {
			reversed = safe_increment_reversed(reversed, forward);
			forward /= 10;
		}
		return reversed;
	}

	public int safe_increment_reversed(int reversed, int forward) {
		reversed *= 10;
		reversed += forward % 10;
		return reversed;
	}
}

File: python/leetcode/is_palindrome/is_palindrome.py
import math

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        if x == 0: return True
        len_n = int(math.log10(x) + 1)
        il = int(len_n/2) - 1
        while il >= 0:
            iu = len_n - il - 1
            if x//10**iu%10 != x//10**il%10: return False
            il -= 1
        return True
